ext-apiserver/pkg/controller/machine/controller.go

```diff
@@ -34,22 +39,94 @@ type MachineControllerImpl struct {
 
 	// lister indexes properties about Machine
 	lister listers.MachineLister
+
+	// lister indexes properties about Cluster
+	clusterLister listers.ClusterLister
+
+	actuator cloud.MachineActuator
+
+	clientSet *clientset.Clientset
 }
 
 // Init initializes the controller and is called by the generated code
 // Register watches for additional resource types here.
 func (c *MachineControllerImpl) Init(arguments sharedinformers.ControllerInitArguments) {
 	// Use the lister for indexing machines labels
 	c.lister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Machines().Lister()
+	c.clusterLister = arguments.GetSharedInformers().Factory.Cluster().V1alpha1().Clusters().Lister()
+
+	clientset, err := clientset.NewForConfig(arguments.GetRestConfig())
+	if err != nil {
+		glog.Fatalf("error creating machine client: %v", err)
+	}
+	c.clientSet = clientset
+
+  // Create machine actuator.
+  // TODO: Assume default namespace for now. Maybe a separate a controller per namesapce?
+	machInterface := clientset.ClusterV1alpha1().Machines(apiv1.NamespaceDefault)
+	var config *cfg.Configuration = &cfg.ControllerConfig
+	actuator, err := cloud.NewMachineActuator(config.Cloud, config.KubeadmToken, machInterface)
+	if err != nil {
+		glog.Fatalf("error creating machine actuator: %v", err)
+	}
+	c.actuator = actuator
 }
 
 // Reconcile handles enqueued messages
-func (c *MachineControllerImpl) Reconcile(u *v1alpha1.Machine) error {
+func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
```

Comment:
Could you add a comment that indicates the delete will be handled by the finalizer?

Reply:
Done